pylint/checkers/classes/class_checker.py

```diff
@@ -1356,6 +1362,33 @@ def _check_slots(self, node):
                     self._check_slots_elt(elt, node)
                 except astroid.InferenceError:
                     continue
+            self._check_redefined_slots(node, slots, values)
+
+    def _check_redefined_slots(
+        self,
+        node: nodes.ClassDef,
+        slots_node: nodes.NodeNG,
+        slots_list: List[nodes.NodeNG],
+    ) -> None:
+        """Check if `node` redefines a slot which is defined in an ancestor class"""
+        slots_names: List[str] = [getattr(slot, "value", "") for slot in slots_list]
+
```

Comment:
I haven't looked at the internals, but how can `slot.value` be nothing?

Reply:
Thanks for the question @DanielNoord. Here is a demo showing when `deque.__name__` has no value attribute:

```
(venv310) cat demo.py 
from collections import deque

class Base:
    __slots__ = "a", "b", "__dict__", deque.__name__
(venv310)
(venv310) pylint demo.py 
> /Users/markbyrne/programming/pylint/pylint/checkers/classes/class_checker.py(1375)_check_redefined_slots()
-> slots_names = [getattr(slot, "value", "") for slot in slots_list]
(Pdb) slots_list
[<Const.str l.3 at 0x10a4a52a0>, <Const.str l.3 at 0x10a4a50f0>, <Const.str l.3 at 0x10a4a5210>, <Attribute.__name__ l.3 at 0x10a4a5240>]
(Pdb) slots_list[3]
<Attribute.__name__ l.3 at 0x10a4a5240>
(Pdb) slots_list[3].value
*** AttributeError: 'Attribute' object has no attribute 'value'
(Pdb) 
```

Reply:
This does answer my question, but creates a follow-up 😄 

Shouldn't we do something with `deque.__name__`? As in, extract `__name__` or something from it? If a subclass redefines `deque.__name__` it should raise as well right?

This `getattr` is a different way of doing `isinstance(slot, nodes.Const)` which I find a bit more explicit. So whatever we decide on for `deque.__name__` personally I would prefer rewriting as a `Const` check.